src/Compilers/CSharp/Test/Semantic/Semantics/StructConstructorTests.cs

```diff
@@ -3765,19 +3767,7 @@ public void FieldInitializer_EscapeAnalysis_04()
     public Example() {}
     static Span<T> F<T>(D<T> d) => d();
 }";
-            var comp = CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular10);
-            comp.VerifyDiagnostics(
-                // (5,31): error CS0188: The 'this' object cannot be used before all of its fields have been assigned. Consider updating to language version '11.0' to auto-default the unassigned fields.
-                //     public Span<byte> Field = F(() => stackalloc byte[512]);
-                Diagnostic(ErrorCode.ERR_UseDefViolationThisUnsupportedVersion, "F").WithArguments("11.0").WithLocation(5, 31),
-                // (5,39): error CS8353: A result of a stackalloc expression of type 'Span<byte>' cannot be used in this context because it may be exposed outside of the containing method
-                //     public Span<byte> Field = F(() => stackalloc byte[512]);
-                Diagnostic(ErrorCode.ERR_EscapeStackAlloc, "stackalloc byte[512]").WithArguments("System.Span<byte>").WithLocation(5, 39),
-                // (6,51): error CS8353: A result of a stackalloc expression of type 'Span<byte>' cannot be used in this context because it may be exposed outside of the containing method
-                //     public Span<byte> Property { get; } = F(() => stackalloc byte[512]);
-                Diagnostic(ErrorCode.ERR_EscapeStackAlloc, "stackalloc byte[512]").WithArguments("System.Span<byte>").WithLocation(6, 51));
-
-            comp = CreateCompilationWithSpan(source);
+            var comp = CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(languageVersion));
             comp.VerifyDiagnostics(
                 // (5,39): error CS8353: A result of a stackalloc expression of type 'Span<byte>' cannot be used in this context because it may be exposed outside of the containing method
```

Comment:
I didn't understand why the definite assignment error disappeared from C# 10. Did we skip it due to presence of ref safety errors? It feels like it would be fine to run both all the time.